search-impl/src/main/java/com/example/elasticsearch/Query.java

```diff
@@ -2,10 +2,12 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Value;
 
 /**
  *
  */
+@Value
 public class Query {
```

Comment:
The classes modelling the API to ElasticSearch don't always need the `@Value` annotation.